segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerKeyIndex.java

```diff
@@ -66,6 +67,18 @@
 @Slf4j
 class ContainerKeyIndex implements AutoCloseable {
     //region Members
+    /**
+     * The maximum number of Table Entries that can exist in the Tail Index for a Table Segment at any given time. The
+     * Tail Index is made of recent updates and removals that have not yet been indexed.
+     *
+     * This value is used for throttling: once a particular Table Segment has reached this number of unindexed entries,
+     * subsequent updates and removals will be blocked (asynchronously) until the {@link IndexWriter} has caught up.
+     *
+     * This value should not be put in a user-configurable location, such as a config file. It is tuned based on the
+     * performance of the {@link IndexWriter}and {@link WriterTableProcessor} to handle indexing tasks and should be updated
+     * as those two classes are improved to enhance indexing speed.
+     */
+    static final int MAX_UNINDEXED_ENTRY_COUNT = 10000;
```

Comment:
Given the issue we saw with the Rocks queue should we make this lower to avoid a similar problem?